### ISSUE TITLE
Empty form fields are missing from the model obgect

### ISSUE THREAD

**Issue by AviFix** [guillotinaweb/ngx-schema-form#384]
Hi,

When leaving a field empty, its property seems to be removed from the form value.

Regular angular form:
![image](https://user-images.githubusercontent.com/3835504/107160815-3680e480-69a1-11eb-83dd-01485a47b89e.png)
firstName is empty but is exist

ngx-schema-form:
![image](https://user-images.githubusercontent.com/3835504/107160829-4dbfd200-69a1-11eb-8c9b-371dc44620c2.png)

email is empty and **not** showing in the model object.

Thanks

**Comment by daniele-pecora** [comment]
That is true, but how could it be handled otherwise and maintain the required validation?

**Comment by AviFix** [comment]
What do you mean?
Does it contradict?

**Comment by daniele-pecora** [comment]
Partially it does, because `required` wouldn't trigger if field is empty.

**Comment by AviFix** [comment]
Isn't it possible to trigger the required if the field is empty like the angular form control does?

**Comment by daniele-pecora** [comment]
ngx-schema-form doesn't rely on angular form control validation rather on a json schema validator.
The default validator implementation is Z-Schema.
JSON Schema validation counts an empty property as present and therefore the required rule is passed as 'valid'.
A 'required' validation is negative only if the required property is not present at all.

**Comment by AviFix** [comment]
Thanks for the detaild explenation.

Could you please point me to the code that is in charge of removing the empty fields and to the validation part?

**Comment by daniele-pecora** [comment]
I don't know it myself.
For preventing a huge breaking change this should end in a feature that can optionally be activated.
Maybe via `forRoot(...)` or smt else

**Comment by AviFix** [comment]
Sure

Anyone else that can point me toi the code that is in charge of removing the empty fields and to the validation?

**Comment by daniele-pecora** [comment]
@AviFix I would guess that `widget.ts` could be a good start (just guessing)
